### PR TITLE
Use setuptools scm versioning

### PR DESCRIPTION
**Pull request by adybbroe**
Use setuptools_scm to manage package versions in SCM (git) metadata instead of declaring them as the version argument or in a git managed file.

### CODE REVIEW

**Comment by adybbroe** [comment]
@mraspaud or @gerritholl Do one of you mind have a quick look at this one?

**Review by djhoese — COMMENTED** [review]
See https://github.com/pytroll/satpy/pull/1512

I would vote for writing to a `aapp_runner/version.py` file instead of going the `pkg_resources` route. There are performance issues when depending on `pkg_resources` and it is a big problem when done on import (like this would be.

**Comment by adybbroe** [comment]
> See [pytroll/satpy#1512](https://github.com/pytroll/satpy/pull/1512)
> 
> I would vote for writing to a `aapp_runner/version.py` file instead of going the `pkg_resources` route. There are performance issues when depending on `pkg_resources` and it is a big problem when done on import (like this would be.

Like so @djhoese ?
Thanks!

**Comment by adybbroe** [comment]
Hmm, so this would require me to add `aapp_runner/version.py` to the repo, and at every new version this file is updated. Have I understood you right @djhoese ?

**Comment by djhoese** [comment]
@adybbroe No! You add `version.py` to the `.gitignore` and **never** commit it. It will be generated when the package is installed.

File: aapp_runner/__init__.py [hunk 1]
```diff
@@ -22,3 +22,6 @@
 
 """Initialise the aapp_runner package
 """
+
+from setuptools_scm import get_version
+version = get_version(root='..', relative_to=__file__)
```
Comment (djhoese) [inline aapp_runner/__init__.py]: I think you should be able to do `from .version import version as __version__`.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[build-system]
+requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4"]
```
Comment (djhoese) [inline pyproject.toml]: I think you want to add the git archive package here too.

File: aapp_runner/__init__.py (hunk 1 above)
Comment (adybbroe) [reply aapp_runner/__init__.py]: Ok, not sure I follow, what is wrong with the above?
I was looking here: https://pypi.org/project/setuptools-scm/

Comment (djhoese) [reply aapp_runner/__init__.py]: Ok. I've never understood that function. I think it is supposed to be used in `setup.py` but then they give examples like the one you're using. If the package is installed then a `<pkg>/version.py` should exist so you can just import the module directly:

```
from .version import version as __version__
```

which would make it possible to do:

```
import aapp_runner
print(aapp_runner.__version__)
```

File: pyproject.toml (hunk 2 above)
Comment (adybbroe) [reply pyproject.toml]: Ok, thanks! Done

File: aapp_runner/__init__.py (hunk 1 above)
Comment (adybbroe) [reply aapp_runner/__init__.py]: Ah, I see, yes, that seem to work, and no need to make other imports!

**Review by djhoese — APPROVED** [review]
Looks good. I think the only concern I have is that for backwards compatibility with older versions of pip I think you could add the `setuptools_scm` options to the `setup.py` too. But given the smaller audience for this package maybe that doesn't matter (pyproject.toml should be good enough).

**Comment by adybbroe** [comment]
> Looks good. I think the only concern I have is that for backwards compatibility with older versions of pip I think you could add the `setuptools_scm` options to the `setup.py` too. But given the smaller audience for this package maybe that doesn't matter (pyproject.toml should be good enough).

Yeah, thanks, I saw that comment also on the setuptools_scm pages, but I thought as you said that with the (still) rather small audience here, it is okay.

**Pull request merged**